concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -1149,13 +1149,9 @@ handleContractUpdateV1 originAddr istance checkAndGetSender transferAmount recei
                                                   euEvents = rrdLogs
                                                 }
                                     in  Right (rrdReturnValue, event : events)
-                            -- execution terminated, commit the new state if it changed
-                            if rrdStateChanged
-                                then do
-                                    -- charge for storing the new state of the contract
-                                    withInstanceStateV1 istance rrdNewState $ \_modifiedIndex -> return result
-                                else do
-                                    return result
+                            if WasmV1.rcFixRollbacks rcConfig || rrdStateChanged
+                                then withInstanceStateV1 istance rrdNewState rrdStateChanged $ \_modifiedIndex -> return result
```

Comment:
Please document this a bit more. At least, we shouldn't remove useful comments that were there previously.

Reply:
Added a bit

This comment
```

                                    -- charge for storing the new state of the contract
```

was wrong so it is removed though.